fdbserver/storageserver.actor.cpp

```diff
@@ -8892,6 +8893,47 @@ ACTOR Future<Void> tryGetRangeForBulkLoad(PromiseStream<RangeResult> results,
 	}
 }
 
+// Utility function to process sample files during bulk load
+// Not critical if we fail to process sample files
+void processSampleFiles(StorageServer* data,
+                        const std::string& bulkLoadLocalDir,
+                        const std::shared_ptr<BulkLoadFileSetKeyMap>& localFileSets) {
+	for (const auto& [range, fileSet] : *localFileSets) {
+		if (fileSet.hasByteSampleFile()) {
+			std::string sampleFilePath = fileSet.getBytesSampleFileFullPath();
+			std::ifstream sampleFile(sampleFilePath);
+			if (sampleFile.is_open()) {
+				std::string line;
+				while (std::getline(sampleFile, line)) {
+					try {
+						// Process each line of the sample file
+						// Format is typically key\tvalue
+						size_t tabPos = line.find('\t');
+						if (tabPos != std::string::npos) {
+							std::string key = line.substr(0, tabPos);
+							std::string value = line.substr(tabPos + 1);
+							// Add to byte sample set only
+							KeyValueRef kv{ StringRef(key), StringRef(value) };
+							data->byteSampleApplySet(kv, invalidVersion);
+						}
+					} catch (Error& e) {
+						TraceEvent(SevWarn, "StorageServerSampleFileProcessingError", data->thisServerID)
+						    .detail("Error", e.what())
+						    .detail("Line", line)
+						    .detail("File", sampleFilePath);
+						// Continue processing other lines
```

Comment:
We do not want to miss any sample metadata because a wrong byte sample data can trigger wrong DD decision which can cause performance bug. We may want to do a full retry here, such as reading all byte sample metadata of a file to memory at first and then write to the SS metadata.

Reply:
Ok. Doing.